src/metadata_validator.py

```diff
@@ -322,6 +345,23 @@ def validate_required_props(self, data_record, msg_prefix):
             if anode_definition["properties"][data_key]["required"]:
                 if data_value is None or not str(data_value).strip():
                     result[ERRORS].append(create_error("M003",[msg_prefix, data_key], data_key, data_value))
+                else:
+                    # validate program name and study name required in CRDCDH-2431.  Both are required properties.
+                    if node_type == "program" and data_key == "program_name":
+                        if not self.program_names: # no program associated with the study
+                            result[WARNINGS].append(create_error("M030", [msg_prefix, self.study_name], data_key, data_value))
+                        else:
+                            matched_val = next((name for name in self.program_names if name.lower() == data_value.lower()), None)
+                            if not matched_val:
+                                result[ERRORS].append(create_error("M028", [msg_prefix, ",".join([f'"{name}"' for name in self.program_names])], data_key, data_value))
+                            else:
+                                data_record[PROPERTIES][data_key] = matched_val
+                    
+                    if node_type in ["study"] and data_key in ["study_name", "clinical_study_name"]:
```

Comment:
Similar to above comment, that study node name can be different among data commons, same as the name property.